fastavro/_six.pyx

```diff
@@ -0,0 +1,99 @@
+# cython: auto_cpdef=True
+
+'''Compatiblity for Python versions.
+
+Some of this code is "lifted" from CherryPy.
+'''
+import sys
+import json
+from sys import stdout
+from struct import unpack
+
+_encoding = 'UTF-8'
+
+if sys.version_info >= (3, 0):
+    from io import BytesIO as MemoryIO
+    xrange = range
+    def py3_btou(n, encoding=_encoding):
+        return n.decode(encoding)
+
+    def py3_utob(n, encoding=_encoding):
+        return bytes(n, encoding)
+
+    def py3_json_dump(obj, indent):
+        json.dump(obj, stdout, indent=indent)
+
+    def py3_iterkeys(obj):
```

Comment:
We should probably update `six.py` to have these `iterkeys` functions and update `writer.py` to use them as well (as they are used in the `pyx` files)

Reply:
Good catch! I'll update this. I made these changes in the `.pyx` files before I knew we needed to retain the `.py` files, so some things got out of sync.

Reply:
Done.